### ISSUE TITLE
Proprietary alternative which doesn't rely on autotrace

### ISSUE THREAD

**Issue by piotr-yuxuan** [fablabnbg/inkscape-centerline-trace#18]
TL;DR: Whilst autotrace is dead, here is a proprietary, gratis, online
tool which does pretty much the same thing:
http://online.rapidresizer.com/tracer.php

---

Hi @fablabnbg,

Thanks guys for this extension, it looks like an amazing plugin!
Unfortunately, the library it relies on, autotrace, has dies more than
fifteen years ago. It's being deleted from GNU Linux distributions,
macOS package manager brew repository, and so on.

However a (rather heroic) attempt to ressurect this project can be
found on GitHub. If this attempt proves successful, do you think you
could use it for this extension?

Further links

- [Usecase for autotrace](https://graphicdesign.stackexchange.com/questions/60107/convert-a-line-drawing-from-raster-to-vector-lines)
- [webpage of the antiquated project](http://autotrace.sourceforge.net/)
- [Numerous bugs of autotrace on current computers](https://graphicdesign.stackexchange.com/questions/48599/average-stroke-from-a-fill/48608#48608)
- [Initial CVS repository of the project](https://sourceforge.net/p/autotrace/code)
- [useful inkscape extension using autotrace, not working because of autotrace death](https://github.com/fablabnbg/inkscape-centerline-trace/)
- [Heroic, benevolent attempt to ressurect this library](https://github.com/autotrace/autotrace) (happy Easter!)

**Comment by jnweiger** [comment]
For me, inkscape-cerntline-trace works quite well.

There is https://github.com/scottvr/autotrace-win64-binaries
and https://sourceforge.net/projects/autotrace/files/AutoTrace/0.31.1/
On my ubuntu machines, a simple 'configure; make; make install' still does the trick with the 0.13.1 release from 2013.

This old release compiles fine without ImageMagick support, which is sufficient for inkscape-centerline-trace.

It is great to see the project is resurrected on github. But I haven't yet succeeded in compiling from there. Apparently it needs some ImageMagick devel files, which I do not have. Let's see, how things  improve. It can only improve from here... :-)

**Comment by piotr-yuxuan** [comment]
Aw, I forgot to say I (have to) use macOS, I haven't found binaries for it. Anyway, good to know it works on GNU Linux :-)

**Comment by piotr-yuxuan** [comment]
Well, I pointed to the proprietary solution for people who need to get
a centreline vectorisation but can't get autotrace in a
straightforward manner. Speaking about centreline vectorisation, I've
found a paper about that which looks appealing:

http://orbit.dtu.dk/files/91435878/522.pdf

```
@inproceedings{mioc2004algorithm,
  title={AN ALGORITHM FOR CENTRELINE EXTRACTION USING NATURAL NEIGHBOUR INTERPOLATION},
  author={Mioc, Darka and Anton, Fran{\c{c}}ois and Dharmaraj, Girija},
  booktitle={XXth ISPRS congress},
  year={2004}
}
```

Do you think the approach described in it could be used for this
extension?

**Comment by jnweiger** [comment]
@piotr-yuxuan thank you for compiling the list of alternatives. The heroic, benevolent effort of @lemenkov has success: We have autotrace back.

I am adding packaging for Mac, Win, Linux, so the installation is now also much easier. Tested on MacOS-10.13 with works fine.

Closing here.
If one of the alternatives is ready for use, please re-open. I'd be happy to support more than one backend in inkscape-centerline-trace.
